### PR TITLE
Add a Spicy function to render bytes as hex string.

### LINKED ISSUE #229: Add library function to print bytes as hex strings
```
module Issue_229_leadingZeroHexadecimalValues;
 leadingZeroHexadecimalValues();

function leadingZeroHexadecimalValues() {
  local acceptable: uint64 = b"36000000000".to_uint();
  print "36000000000 rendered %s0X and %s010X: %0X %010X" % ("%", "%", acceptable, acceptable);
  acceptable = 400;
  print "400 rendered %sX and %s0X: %X %0X" % ("%", "%", acceptable, acceptable);
}
```
shows that %0X is not a specifier which does that.
```
3470 /usr/local/etc/tmp$spicy/build/bin/spicyc -j Issue_229_leadingZeroHexadecimalValues.spicy
36000000000 rendered %0X and %010X: 861C46800 0861C46800
400 rendered %X and %0X: 190 190
```

### PR DESCRIPTION
**Pull request by bbannier**
Closes #229.

### CODE REVIEW

**Comment by rsmmr** [comment]
Looks like we need to prefix the `__library_type`s in spicy.spicy with ``::``, e.g.:

 `__library_type("spicy::rt::zlib::Stream")` -> `__library_type("::spicy::rt::zlib::Stream")`

**Comment by rsmmr** [comment]
Turns out the the compiler is using/outputting a few library types as well that now need the new global prefix. I fixed that.

Now looking into compile times.

**Comment by rsmmr** [comment]
See #301 on compile times.

**Comment by rsmmr** [comment]
@bbannier have a look

File: spicy/src/rt/util.cc [hunk 1]
```diff
@@ -18,3 +19,12 @@ std::string spicy::rt::version() {
 #error "Neither HILTI_RT_BUILD_TYPE_DEBUG nor HILTI_RT_BUILD_TYPE_RELEASE define."
 #endif
 }
+
+std::string spicy::rt::bytes_to_hexstring(const hilti::rt::Bytes& value) {
+    std::string result;
+
+    for ( auto x : value )
+        result += hilti::rt::fmt("%02X", x);
```
Comment (bbannier) [inline spicy/src/rt/util.cc:27]: Nice, this should be a bit lighter than my Spicy impl.

**Pull request merged**